stdlib/src/collections/counter.mojo

```diff
@@ -59,6 +59,32 @@ struct Counter[V: KeyElement](Sized, CollectionElement, Boolable):
             var item = item_ref[]
             self._data[item] = self._data.get(item, 0) + 1
 
+    @always_inline
+    fn __init__(inout self, *, other: Self):
+        """Create a new Counter by copying another Counter.
+
+        Args:
+            other: The Counter to copy.
+        """
+        self._data = Dict[V, Int](other=other._data)
+
+    @staticmethod
+    fn fromkeys(keys: List[V], value: Int) -> Self:
```

Comment:
Then we need a customisation point based on the numerical type `V`, e.g. saturated subtraction for unsigned types.

Reply:
Actually, Python's `Counter.frommkeys` doesn't support for it can be ambiguous, should we add this at all?

> ```python
> # There is no equivalent method for counters because the semantics
> # would be ambiguous in cases such as Counter.fromkeys('aaabbc', v=2).
> # Initializing counters to zero values isn't necessary because zero
> # is already the default value for counter lookups.  Initializing
> # to one is easily accomplished with Counter(set(iterable)).  For
> # more exotic cases, create a dictionary first using a dictionary
> # comprehension or dict.fromkeys().
> ```

Reply:
> What should be the behavior if `Int` is negative? I'm good with debug_assert or whatever check, as long as we have at least something.
> 
> Also, a wild idea: could `Counter` be implemented with a value type parameter? That way we could have `UInt` or even `UInt8` counters.

I like your crazy idea. Let's explore it in a separate PR. Is it possible to have a default param value in Mojo, like not having to declare the counter like `Counter[Int]` if the default count type is `Int`? If not, we would not be honoring the Python `Counter`, so any Python code that uses `Counter` should adapt its constructors

Reply:
> @msaelices Could you please add a debug assert or ensure some other way that the value is not negative?

Done: https://github.com/msaelices/mojo/commit/df5b4f7270fe22735054d865fc0d0f1f118d4c11